agent/engine/dockerclient/dockerclientfactory.go

```diff
@@ -106,6 +110,18 @@ func (f *factory) FindKnownAPIVersions() []DockerVersion {
 	return knownVersions
 }
 
+// FindClientAPIVersion returns the version of the client from the map
+// TODO we should let go docker client return this version information
```

Comment:
the TODO seems like a better option, whats stopping us from doing that?

Reply:
go dockerclient doesn't support this today, we need contribute and update.